src/matlab2tikz.m

```diff
@@ -3205,10 +3205,23 @@ case size(cData,2) % specific y-data is given
     prop         = switchMatOct(m2t, 'BarPeers', 'bargroup');
     bargroup     = get(h, prop);
     numBarSeries = numel(bargroup);
+    
     if isHG2(m2t)
+        % In HG2, BarPeers are sorted in reverse order wrt HG1
         bargroup = bargroup(end:-1:1);
+    
+    elseif strcmpi(m2t.env,'MATLAB')
+        % In HG1, h is a double but bargroup a graphic object. Cast h to a
+        % graphic object
+        h = handle(h);
+    
+    else
+        % In Octave, the bargroup is a replicated cell array. Pick first 
+        bargroup = bargroup{1};
     end
-    [~, barSeriesId] = ismember(handle(h), bargroup);
+    
+    % Get bar series Id
+    [~, barSeriesId] = ismember(h, bargroup);
```

Comment:
The PR https://github.com/matlab2tikz/matlab2tikz/pull/537 was accomplishing this. Once that PR was in place, I just rebased this PR (and solved the conflict).